src/screens/Survey/SurveyMainScreen.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { Text, View, StyleSheet, Button} from 'react-native';
+import {createStackNavigator} from 'react-navigation';
+import SurveyStart from './SurveyStart';
+import Question from './redux/containers/Container_Question';
+
+var RootStack = createStackNavigator(
+  {
+    Home: {screen: SurveyStart},
+    Question: {screen: Question},
+  },
+  {
+    cardStyle : {
+      backgroundColor: 'transparent',
+      alignItems:'center',
+      justifyContent:'center'
+    },
+    headerMode: 'none',
+    mode: 'card',
+    swipeEnabled: true,
+    navigationOptions: () => ({
+      gesturesEnabled : true,
+      gestureResponseDistance: {
+        horizontal: 200
+      }
+    }),
+    transitionConfig: () => ({
+            screenInterpolator: sceneProps => {
+                const { layout, position, scene } = sceneProps;
+                const { index } = scene;
+
+                const translateX = position.interpolate({
+                    inputRange: [index - 1, index, index + 1],
+                    outputRange: [layout.initWidth, 0, 0]
+                });
+
+                const opacity = position.interpolate({
+                    inputRange: [index - 1, index - 0.99, index, index + 0.99, index + 1],
+                    outputRange: [0, 1, 1, 0.3, 0]
+                });
+
+                return { opacity, transform: [{ translateX }] }
+            }
+      })
+  }
+);
+
+ class SurveyMainScreen extends React.Component {
+
+   constructor(props){
+     super(props);
+   }
+
+  componentDidMount() {
+    this.props.loadSurvey();
+   }
+
+   componentDidUpdate(){
+     Object.keys(this.props.surveyQuestions).map((item,index)=>{
+       this.props.addKeyToSet(item)
+     })
+   }
+
+  render() {
+
+    return (
+      <View style={{
+        flex: 1,
+        flexDirection: 'column',
+        justifyContent: 'space-between',
+      }}>
+
+        <View style={styles.container}>
+          <View style={[styles.survey_block]}>
+            <RootStack screenProps={{rootNav: this.props.rootnav}}/>
```

Comment:
I still think it's weird to have a navigation stack inside of a screen. Usually all of your screens are inside the navigation stack

Reply:
Ok, I think I fixed this. I basically emulated the same solution they had in the react navigation documentation